homeassistant/components/sensor/darksky.py

```diff
@@ -188,6 +203,17 @@ def unit_system(self):
         """Return the unit system of this entity."""
         return self.forecast_data.unit_system
 
+    @property
+    def entity_picture(self):
+        """Return the entity picture to use in the frontend, if any."""
+        if self._icon is None or "summary" not in self.type:
```

Comment:
You set icon to an empty string . So instead you should check just `if self._icon`